src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -6,62 +6,67 @@
 from lxml import etree
 from viztracer import VizTracer
 
-from dsp_tools.analyse_xml_data.models import ResptrLink, UploadResource, XMLLink
+from dsp_tools.analyse_xml_data.models import ResptrLink, XMLLink
 
 
-def _create_info_from_xml_for_graph(root: etree._Element) -> tuple[list[ResptrLink], list[XMLLink], set[str]]:
+def _create_info_from_xml_for_graph(
+    root: etree._Element,
+) -> tuple[etree._Element, list[ResptrLink], list[XMLLink], list[str]]:
     """Create instances of the classes ResptrLink and XMLLink from the root of the XML file."""
```

Comment:
You return 4 things, but only 2 of them are described here. 
 1. Can you reduce the number of things you return? 
 2. Can you describe everything that you return?

Reply:
there is a simple way of returning fewer things: group them into an object. :)  
I know to some people this feels like adding overhead - but for me as a functional programmer, this adds type safety, which is a good thing.

Reply:
The root is now removed, so that reduces it to 3. For me in this case it feels like overkill. I create an object at this point to return it. And after it is returned I immediately unpack it again...